oauth2_http/java/com/google/auth/oauth2/GoogleCredentials.java

```diff
@@ -331,6 +334,74 @@ public GoogleCredentials createWithQuotaProject(String quotaProject) {
     return this.toBuilder().setQuotaProjectId(quotaProject).build();
   }
 
+  @VisibleForTesting
+  TrustBoundary getTrustBoundary() {
+    return trustBoundary;
+  }
+
+  /**
+   * Returns whether the credentials support trust boundary.
+   *
+   * @return {@code true} if the credentials support trust boundary, {@code false} otherwise.
+   */
+  boolean supportsTrustBoundary() {
+    return false;
+  }
+
+  /**
+   * Refreshes the trust boundary by making a call to the trust boundary URL.
+   *
+   * <p>This method is for internal use only and should not be called by users directly. It is used
+   * to enforce security policies by ensuring that the credentials used to access Google Cloud APIs
+   * are not used outside a trusted environment.
+   *
+   * @param newAccessToken The new access token to be used for the refresh.
+   * @param trustBoundaryUrl The URL of the trust boundary service.
+   * @param transportFactory The HTTP transport factory to be used for the refresh.
+   * @throws IOException If the refresh fails and no cached value is available.
+   */
+  @InternalApi
+  void refreshTrustBoundary(
```

Comment:
Multiple threads might read trustBoundary as null inside the first synchronized block, and then all proceed to call TrustBoundary.refresh() simultaneously, resulting in a 'thundering herd' of redundant requests to the TB endpoint.

Do we need to implement request coalescing here (similar to how OAuth2Credentials uses refreshTask for access tokens) so that only one network request is in flight at a time? 


cc @lqiu96 since I think you guys might've already discussed this